src/Compiler/CompilerOptions.cpp

```diff
@@ -281,6 +282,19 @@ static llvm::cl::opt<std::string, true> shapeInformationOpt("shapeInformation",
     llvm::cl::value_desc("value"), llvm::cl::location(shapeInformation),
     llvm::cl::cat(OnnxMlirOptions));
 
+static llvm::cl::opt<std::string, true> dimParamsOpt("dimParams",
+    llvm::cl::desc(
+        "Custom onnx.dim_params attributes for the inputs of the ONNX model for"
+        "specifying relationship among ranks of the inputs.\n"
+        "\"value\" is in the format of "
+        "\"INPUT_ID1:D1=S1,D2=S2,...,Dn=Sn|INPUT_ID2:D1=T1,D2=T2,...Dn=Tn\""
+        "where \"INPUT_ID1, INPUT_ID2, ...\" are input indices (starting from "
+        "0 or being -1 for all input indices), and\n"
+        "\"S1, S2, ...\" and \"T2, T2, ...\" are symbols to specify that same "
+        "symbols have the same value"),
```

Comment:
I would change this to make it more clear.

`
"\"S1, S2, ...\" and \"T2, T2, ...\" are symbols assigned to each unknown dimension. Two dimensions with the same symbols are expected to have the same value at runtime"),
`

BTW, what is the valid format of symbols? arbitrary string?

Reply:
Should we say that `this custom onnx.dim_params will override the existing onnx.dim_params if exists`?

Reply:
I updated the help messages to clarify the meaning of the option, and comments for handing this option in src/Builder/FrontendDialectTransformer.cpp . Please review them again. Thanks.